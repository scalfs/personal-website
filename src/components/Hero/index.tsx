import Transition from 'components/Transition'

const Hero = () => {
  return (
    <section className="flex h-screen">
      <Transition timeout={750} type="none">
        <div className="my-auto max-w-xl space-y-4 text-justify text-xl sm:text-2xl">
          <Transition isChild type="fadeup">
            <h1>Hi, there! My name is</h1>
          </Transition>
          <Transition isChild delay={150} type="fadeup">
            <h2 className="bg-gradient-to-r from-[#4586e2] to-[#214bc1] bg-clip-text text-6xl font-extrabold leading-none tracking-tight text-transparent sm:text-[5rem]">
              Vitor Escalfoni.
            </h2>
          </Transition>
          <Transition isChild delay={300} type="fadeup">
            <h3>
              I&apos;m a <span className="font-medium">software engineer</span>{' '}
              specialized in{' '}
              <span className="font-medium">mobile app development,</span>{' '}
              passionate about crafting great user experiences and implementing
              beautiful designs.
            </h3>
          </Transition>
        </div>
      </Transition>
    </section>
  )
}

export default Hero
