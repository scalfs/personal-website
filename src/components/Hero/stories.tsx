import { ComponentMeta, ComponentStory } from '@storybook/react'
import Hero from '.'

export default {
  title: 'Hero',
  component: Hero
} as ComponentMeta<typeof Hero>

const Story: ComponentStory<typeof Hero> = () => <Hero />

export const Default = Story.bind({})
