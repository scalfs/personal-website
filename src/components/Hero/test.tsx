import { render, screen } from '@testing-library/react'

import Hero from '.'

describe('<Hero />', () => {
  it('should render the heading', async () => {
    const { container } = render(<Hero />)

    expect(
      await screen.findByRole('heading', { name: /Vitor Escalfoni/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
