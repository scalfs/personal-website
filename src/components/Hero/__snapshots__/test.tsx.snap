// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render the heading 1`] = `
<section
  class="flex h-screen"
>
  <div
    class="my-auto max-w-xl space-y-4 text-justify text-xl sm:text-2xl"
  >
    <div
      class="duration-300 ease-in-out-cubic motion-reduce:transform-none opacity-0 translate-y-4"
      style="transition-delay: undefinedms;"
    >
      <h1>
        Hi, there! My name is
      </h1>
    </div>
    <div
      class="duration-300 ease-in-out-cubic motion-reduce:transform-none opacity-0 translate-y-4"
      style="transition-delay: 150ms;"
    >
      <h2
        class="bg-gradient-to-r from-[#4586e2] to-[#214bc1] bg-clip-text text-6xl font-extrabold leading-none tracking-tight text-transparent sm:text-[5rem]"
      >
        Vitor Escalfoni.
      </h2>
    </div>
    <div
      class="duration-300 ease-in-out-cubic motion-reduce:transform-none opacity-0 translate-y-4"
      style="transition-delay: 300ms;"
    >
      <h3>
        I'm a 
        <span
          class="font-medium"
        >
          software engineer
        </span>
         
        specialized in
         
        <span
          class="font-medium"
        >
          mobile app development,
        </span>
         
        passionate about crafting great user experiences and building
         
        <span
          class="font-medium"
        >
          outstanding products
        </span>
        .
      </h3>
    </div>
  </div>
</section>
`
